File: laravel/app/Models/Task.php
<?php

/**
 * Created by Reliese Model.
 */

namespace App\Models;

use Illuminate\Database\Eloquent\Collection;
use Illuminate\Database\Eloquent\Model;

/**
 * Class Task
 * 
 * @property uuid $Id
 * @property string $Name
 * @property string|null $Description
 * @property int $TaskTypeId
 * @property int $TaskStatusId
 * @property uuid $DeveloperId
 * @property uuid|null $SprintId
 * 
 * @property TaskType $task_type
 * @property TaskStatus $task_status
 * @property User $user
 * @property Sprint|null $sprint
 * @property Collection|TaskHistory[] $task_histories
 *
 * @package App\Models
 */
class Task extends Model
{
	protected $table = 'Tasks';
	protected $primaryKey = 'Id';
	public $incrementing = false;
	public $timestamps = false;

	protected $casts = [
		'Id' => 'uuid',
		'TaskTypeId' => 'int',
		'TaskStatusId' => 'int',
		'DeveloperId' => 'uuid',
		'SprintId' => 'uuid'
	];

	protected $fillable = [
		'Name',
		'Description',
		'TaskTypeId',
		'TaskStatusId',
		'DeveloperId',
		'SprintId'
	];

	public function task_type()
	{
		return $this->belongsTo(TaskType::class, 'TaskTypeId');
	}

	public function task_status()
	{
		return $this->belongsTo(TaskStatus::class, 'TaskStatusId');
	}

	public function user()
	{
		return $this->belongsTo(User::class, 'DeveloperId');
	}

	public function sprint()
	{
		return $this->belongsTo(Sprint::class, 'SprintId');
	}

	public function task_histories()
	{
		return $this->hasMany(TaskHistory::class, 'TaskId');
	}
}

File: JavaService/src/main/java/org/student/microserviceapp/javaservice/models/Task.java
package org.student.microserviceapp.javaservice.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "\"Tasks\"", schema = "public", indexes = {
        @Index(name = "IX_Tasks_TaskTypeId", columnList = "TaskTypeId"),
        @Index(name = "IX_Tasks_TaskStatusId", columnList = "TaskStatusId"),
        @Index(name = "IX_Tasks_DeveloperId", columnList = "DeveloperId"),
        @Index(name = "IX_Tasks_SprintId", columnList = "SprintId")
})
public class Task {
    @Id
    @Column(name = "\"Id\"", nullable = false)
    private UUID id;

    @Column(name = "\"Name\"", nullable = false, length = Integer.MAX_VALUE)
    private String name;

    @Column(name = "\"Description\"", length = Integer.MAX_VALUE)
    private String description;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "\"TaskTypeId\"", nullable = false)
    private TaskType taskType;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "\"TaskStatusId\"", nullable = false)
    private TaskStatus taskStatus;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "\"DeveloperId\"", nullable = false)
    private User developer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "\"SprintId\"")
    private Sprint sprint;

}